lib/common/utils.c

```diff
@@ -609,6 +610,24 @@ compare_version(const char *version1, const char *version2)
     return rc;
 }
 
+gboolean
+valid_version_format(const char *version)
+{
+    regex_t regex;
+    int rc = 0;
+    const char *format_regex = "^[[:digit:]]+([.][[:digit:]]+)*$";
+
+    if (version == NULL) {
+        return FALSE;
+    }
+
+    regcomp(&regex, format_regex, REG_EXTENDED);
```

Comment:
The easiest way is to make regex static here, and initialize it once. I don't think there's a portable way to check whether it has already been initialized, so you'd need a static boolean variable to track that.